cle/loader.py

```diff
@@ -781,7 +783,7 @@ def _map_object(self, obj):
         if obj.has_memory:
             l.info("Mapping %s at %#x", obj.binary, base_addr)
             self.memory.add_backer(base_addr, obj.memory)
-            key_bisect_insort_left(self.all_objects, obj, keyfunc=lambda o: o.min_addr)
+        key_bisect_insort_left(self.all_objects, obj, keyfunc=lambda o: o.min_addr)
```

Comment:
I'm worried about this change. Does `o.min_addr` always hold a meaningful value if `o.has_memory` is False?

Reply:
If it doesn't, it should.....?

Reply:
The Soot backend has `has_memory` set to False. In that case, any "memory-address-related" properties (including `mapped_base`, `min_addr`, `max_addr`, etc.) won't be valid.

I think `has_memory` means "if this object is mapped to a integer-indexed memory region." In that case, `has_memory` should be set to `True` for your `NamedRegion` as well, and we don't need to change this line.

Reply:
Not quite, if you set `has_memory` to True here, as the code stands today, NamedRegion will need something approximating a Clemory back there. This approach seemed much cleaner. We aren't faking memory where there isn't any

Reply:
> NamedRegion will need something approximating a Clemory back there

Why so? `NamedRegion` can simply implement `min_addr` and `max_addr`.

Reply:
Because in order to call add_backer(), you need a Clemaory.  We can't put real memory there, because these regions will be *gigantic*.  We either need a fake Clemory or to not do it this way

Reply:
Now that I'm thinking about it, I think a better way going forward is to change the definition of `has_memory` to "whether this backend is backed by a Clemory object or not." `min_addr` and `max_addr` should always be something meaningful so that the backend object (even if it's a Soot object) can always be added to `all_objects`.

I opened a PR to reflect this new definition: https://github.com/angr/cle/pull/195. It also includes this line of change.